cmds/core/wget/wget_test.go

```diff
@@ -40,120 +43,108 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-var tests = []struct {
-	name    string
-	flags   []string // in, %[1]d is the server's port, %[2] is an unopen port
-	url     string   // in
-	content string   // out
-	retCode int      // out
-}{
-	{
-		name:    "basic",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	{
-		name:    "ipv4",
-		flags:   []string{},
-		url:     "http://127.0.0.1:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	// TODO: CircleCI does not support ipv6
-	// {
-	// 	name:    "ipv6",
-	// 	flags:   []string{},
-	// 	url:     "http://[::1]:%[1]d/200",
-	// 	content: content,
-	// 	retCode: 0,
-	// },
-	{
-		name:    "redirect",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/302",
-		content: "",
-		retCode: 0,
-	},
-	{
-		name:    "4xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/404",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "5xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/500",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "no server",
-		flags:   []string{},
-		url:     "http://localhost:%[2]d/200",
-		content: "",
-		retCode: 1,
-	},
-}
-
-func getListener(t *testing.T) (net.Listener, int) {
-	t.Helper()
-	l, err := net.Listen("tcp", ":0")
+func getUnusedPort() (int, error) {
+	a, err := net.ResolveTCPAddr("tcp", ":0")
 	if err != nil {
-		t.Fatalf("error setting up TCP listener: %v", err)
+		return 0, err
 	}
-	return l, l.Addr().(*net.TCPAddr).Port
+	l, err := net.ListenTCP("tcp", a)
+	if err != nil {
+		return 0, err
+	}
+
+	l.Close()
+	return l.Addr().(*net.TCPAddr).Port, nil
 }
 
 // TestWget implements a table-driven test.
 func TestWget(t *testing.T) {
-	// Start a webserver on a free port.
-	l, port := getListener(t)
-	defer l.Close()
-	ul, unusedPort := getListener(t)
-	defer ul.Close()
-	go func() {
-		for {
-			conn, err := ul.Accept()
-			if err != nil {
-				// End of test.
-				return
-			}
-			conn.Close()
-		}
-	}()
+	srv := httptest.NewServer(handler{})
+	defer srv.Close()
+	tmpDir := t.TempDir()
+	freePort, err := getUnusedPort()
+	if err != nil {
+		t.Fatal(err)
+	}
 
-	h := handler{}
-	go func() {
-		log.Print(http.Serve(l, h))
-	}()
+	var tests = []struct {
+		name       string
+		url        string // in
+		content    string // out
+		outputPath string
+		err        error
+	}{
+		{
+			name:       "ipv4",
+			url:        fmt.Sprintf("%s/200", srv.URL),
+			content:    content,
+			outputPath: filepath.Join(tmpDir, "basic"),
+			err:        nil,
+		},
+		{
+			name:       "localhost",
+			url:        strings.Replace(srv.URL, "127.0.0.1", "localhost", 1) + "/200",
```

Comment:
this (`127.0.0.1` -> localhost) will break when we change to IPv4 right? Is there any value to this test? What `localhost` means is specific to the OS, DNS resolvers / `hosts` file, etc right? Maybe we should get rid of the 'localhost' tests?

Reply:
Yeah, I'm not quite getting what this test was trying to accomplish, but it was there before so probably original author know more, so to be on the save side it probably will not harm if it will stay for a while.